rails/etsy_product_catalog/app/models/listing.rb

```diff
@@ -1,6 +1,9 @@
 class Listing < ActiveRecord::Base
- has_many :images, foreign_key: :etsy_listing_id
-
+ has_many :images, foreign_key: "etsy_listing_id"
+ #belongs_to :category, class_name: Category
 
+ 	def main_image
+ 		Image.where(etsy_listing_id: etsy_listing_id).first.url_fullxfull
```

Comment:
Since you already have the association set up, use it here. Also, if the method is returning a *url* (rather than a full image object), change the name to match:
```
def main_image_url
  images.first.url_fullxfull
end
```